commit/plugin.go

```diff
@@ -345,6 +362,8 @@ func (p *Plugin) Close() error {
 	if err := p.ccipReader.Close(ctx); err != nil {
 		return fmt.Errorf("close ccip reader: %w", err)
 	}
+
+	p.bgSyncCancelFunc()
```

Comment:
If you want to wait for the syncing goroutine to finish put a `wg.Wait()` here, otherwise you can remove the wg.